### ISSUE TITLE
Bot doesn't ignore non-humanity bots sometimes

### ISSUE THREAD

**Issue by turt2live** [turt2live/CAH_BOTZIE#27]
Sample case:
```
<@jkcclemens> !start
<SirDerpington> A new game is starting!
<SirDerpington> Card packs for this game: v3
<SirDerpington> 45 seconds remain to join the game!
<SirDerpington> Card counts: 90 unused/90 black cards, 460 unused/460 white cards
<SirDerpington> Use !join to join.
<SirDerpington> j​kcclemens has joined the game!
<SirDerpington> The host is j​kcclemens.
<@jkcclemens> ~join
<CahBotzie_Dev> !join
<ctest_a> !join
<SirDerpington> CahBotzie_Dev has joined the game!
<SirDerpington> c​test_a has joined the game!
<@jkcclemens> !start
<SirDerpington>  
<SirDerpington> Round 1!
<SirDerpington> j​kcclemens is the card czar.
<SirDerpington> What ended my last relationship? <BLANK>
<@jkcclemens> heh
<@jkcclemens> two games at once
<@jkcclemens> heh
<@jkcclemens> ~p 1
<@jkcclemens> ~help
<@jkcclemens> uh
<@jkcclemens> did I break your bot because two games
```

The bot broke once it got the notice from `SirDerpington` for it's cards, but didn't track the black card, so it errored out. The bot should consider who is sending the notice.

Got this exception:
```
c:\Git\CAH_BOTZIE\node_modules\irc\lib\irc.js:759
                    throw err;
                          ^
TypeError: Cannot read property 'match' of null
    at Client.<anonymous> (c:\Git\CAH_BOTZIE\build\bot\IrcHumanityGame.js:81:40)
    at Client.emit (events.js:118:17)
    at Client.<anonymous> (c:\Git\CAH_BOTZIE\node_modules\irc\lib\irc.js:245:22)
    at Client.emit (events.js:107:17)
    at iterator (c:\Git\CAH_BOTZIE\node_modules\irc\lib\irc.js:756:22)
    at Array.forEach (native)
    at TLSSocket.<anonymous> (c:\Git\CAH_BOTZIE\node_modules\irc\lib\irc.js:752:15)
    at TLSSocket.emit (events.js:107:17)
    at readableAddChunk (_stream_readable.js:163:16)
    at TLSSocket.Readable.push (_stream_readable.js:126:10)
    at TCP.onread (net.js:529:20)
```

**Comment by turt2live** [comment]
Add'l info:

The crash seems to have been caused by the normal bot, CardAficionado, being "removed" but not disabled. CardAficionado thought it was still playing and CAH_BOTZIE was tracking it, and when it received a notice from CardAficionado it attempted a play on a game it cannot see.

This issue expands to ensuring game state is correct before doing something (ie: is there a black card we can guess on?). Errors may want to be reported to the channel if in debug mode.

**Comment by turt2live** [comment]
Fixed in 2518c4dad3ecbd0b189e33cdd4964b4d697d5984
